frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/ShareButton/utils.tsx

```diff
@@ -87,20 +87,105 @@ export const loadStateFromQueryParams = ({
 }): LoadStateFromURLPayload | null => {
   if (isSSR()) return null;
 
-  const paramsToRemove = [];
+  const paramsToRemove: string[] = [];
 
   // Checks if the URL has any query params related to shared filter state
   // If so, it will update the state with the values from the URL
 
-  // Check for version
+  const { delimiter } = getMetadataForShareLinkParsing({
+    params,
+    paramsToRemove,
+  });
+
+  // Check for filter properties
+  const newSelectedFilters = getNewSelectedFilters({
+    tissues,
+    params,
+    paramsToRemove,
+    selectedFilters,
+    delimiter,
+  });
+
+  // Check for genes
+  const newSelectedGenes = params.get("genes")?.split(delimiter) || [];
+  if (newSelectedGenes.length > 0) paramsToRemove.push("genes");
+
+  // If there are no filters and genes selected, don't update the state
+  if (
+    Object.values(Object.keys(newSelectedFilters)).length === 0 &&
+    newSelectedGenes.length === 0
+  ) {
+    return null;
+  }
+
+  // Check for organism
+  const newSelectedOrganism = params.get("organism") || HUMAN_ORGANISM_ID;
+
+  if (newSelectedOrganism) {
+    paramsToRemove.push("organism");
+  }
+
+  // Check for cell types
+  const newFilteredCellTypes = params.get("cellTypes")?.split(delimiter) || [];
+
+  if (newFilteredCellTypes.length > 0) paramsToRemove.push("cellTypes");
+
+  // Check for compare
+  const newCompare = (params.get("compare") as CompareId) || undefined;
+
+  if (newCompare) {
+    paramsToRemove.push("compare");
+  }
```

Comment:
move checks for these below guard clause to avoid unnecessary computation:

```ts
// If there are no filters and genes selected, don't update the state
  if (
    Object.values(Object.keys(newSelectedFilters)).length === 0 &&
    newSelectedGenes.length === 0
  ) {
    return null;
  }
```